tenant_users/tenants/tasks.py

```diff
@@ -43,11 +51,20 @@ def provision_tenant(tenant_name, tenant_slug, user_email):
         )
 
         # Add user as a superuser inside the tenant
-        tenant.add_user(user, is_superuser=True)
-    except:
-        if tenant is not None:
-            #Flag is set to auto-drop the schema for the tenant
-            tenant.delete(True)
-        raise
+    else:  # django-tenants
+        tenant = TenantModel.objects.create(name=tenant_name,
+                                            slug=tenant_slug,
+                                            schema_name=schema_name,
+                                            owner=user)
+
+        # Add one or more domains for the tenant
+        domain = get_tenant_domain_model().objects.create(domain=tenant_domain,
+                                                          tenant=tenant,
+                                                          is_primary=True)
+
+    tenant.add_user(user, is_superuser=True, is_staff=is_staff)
+    # if tenant is not None:
+    #     #Flag is set to auto-drop the schema for the tenant
+    #     tenant.delete(True)
```

Comment:
What is the reasoning behind having this commented out and having the create code removed from the try: except clause?